src/protego.py

```diff
@@ -0,0 +1,386 @@
+import re
+from datetime import time
+from collections import namedtuple
+import six
+from six.moves.urllib.parse import urlparse, urlunparse, ParseResult, quote, unquote
+import logging
+
+logger = logging.getLogger(__name__)
+
+Rule = namedtuple('Rule', ['field', 'value'])
+RequestRate = namedtuple(
+    'RequestRate', ['requests', 'seconds', 'start_time', 'end_time'])
+
+DISALLOW_DIRECTIVE = {'disallow', 'dissallow', 'dissalow', 'disalow', 'diasllow', 'disallaw'}
+ALLOW_DIRECTIVE = {'allow'}
+USER_AGENT_DIRECTIVE = {'user-agent', 'useragent', 'user agent'}
+SITEMAP_DIRECTIVE = {'sitemap', 'sitemaps', 'site-map'}
+CRAWL_DELAY_DIRECTIVE = {'crawl-delay', 'crawl delay'}
+REQUEST_RATE_DIRECTIVE = {'request-rate', 'request rate'}
+HOST_DIRECTIVE = {'host'}
+
+WILDCARDS = {'*', '$'}
+
+__all__ = ['RequestRate', 'Protego']
```

Comment:
This only affects which objects are imported with `from protego import *`.

If these 2 are the only objects that should be used by end-users, you might want to consider prefixing with `_` all other objects, including `Rule` and constants like `DISALLOW_DIRECTIVE` above.

Otherwise, future releases will need to cover changes to them as backward incompatible (API-breaking) changes.